src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -340,5 +348,15 @@ internal void ValidateTimerCallback(ActorHost host, string callback)
                 throw new ArgumentException("Timer callback can only return type Task");
             }
         }
+
+        private void EnsureInteractorInitialized()
+        {
+            if (this.Host.DaprInteractor == null)
+            {
+                throw new InvalidOperationException(
+                    "The actor was initialized without an HTTP client, and so cannot interact with timers or reminders. " +
+                    "This is likely to happen inside a unit test.");
```

Comment:
As described in the issue, this was already impossible to unit test. Now it just fails with a reasonable error message. I logged #575 to track improving this.